### PR TITLE
♻️Move VariableService to doc level #2

### LINKED ISSUE #22217: 🐛Analytics vars refactor breaks shadow dom
In my refactor #22137 to move variable service to the document level shadow root was not handled correctly. This causes amp analytics in shadow mode to throw and not send analytics pings.

Plan to fix this, and hopefully prevent similar issues in the future:
- [x] Revert #22137 
- [x] Add support for shadow root testing in integration tests
- [x] Create shadow root test for amp-analytics
- [ ] Reimplement variable service refactor

### PR DESCRIPTION
**Pull request by calebcordry**
Reimplementing this refactor after adding shadow integration test. 

Closes #22217

### CODE REVIEW

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -711,8 +713,8 @@ AMP.extension(TAG, '0.1', AMP => {
   AMP.registerServiceForDoc(
       'amp-analytics-instrumentation', InstrumentationService);
   AMP.registerServiceForDoc('activity', Activity);
-  installVariableService(AMP.win);
   installLinkerReaderService(AMP.win);
+  AMP.registerServiceForDoc('amp-analytics-variables', VariableService);
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/amp-analytics.js]: This is where the previous PR was broken. It was written as `installVariableServiceForDoc(AMP.ampdoc);` and `AMP.ampdoc` was not doing the right thing in the shadow case.

Comment (torch2424) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Thanks for the heads up 😄

Comment (lannka) [reply extensions/amp-analytics/0.1/amp-analytics.js]: so should we remove `installVariableServiceForDoc` if it's dangerous?

Comment (calebcordry) [reply extensions/amp-analytics/0.1/amp-analytics.js]: I kind of did? It is now `installVariableServiceForTesting` and only used in tests.

**Comment by mattwomple** [comment]
@calebcordry Did you test this with the original example I shared with you in #22137 ([direct link to comment](https://github.com/ampproject/amphtml/pull/22137#pullrequestreview-234829387))? When I navigate to the second page in the PWA, I see error: `Expected service amp-analytics-variables to be registered`. Full stack trace:

    log.js:184 [Resource] failed to build: amp-analytics#1 Error: Expected service amp-analytics-variables to be registered
        at Log.assert (log.js:342)
        at devAssert (log.js:752)
        at getServiceInternal (service.js:383)
        at getServiceForDoc (service.js:245)
        at variableServiceForDoc (variables.js:317)
        at AmpAnalytics.buildCallback (amp-analytics.js:124)
        at custom-element.js:513
        at new Promise (<anonymous>)
        at HTMLElement.build (custom-element.js:510)
        at Resource.build (resource.js:337)

**Comment by calebcordry** [comment]
@mattwomple I was just going to ask you to test, I tested a few different shadow viewers but I did not see this issue. Admittedly I did not try your example. Looking now.

**Pull request merged**